src/main.ts

```diff
@@ -148,212 +98,108 @@ export class Content {
+      var fileName = searchResult[index].name; 
+      var element = searchResult[index].path;
+      var repoName = searchResult[index].repository.name;
+      var repoOwner = searchResult[index].repository.owner.login;
+
+      // Push file to action list if filename matches action.yaml or action.yml
+      // Search result will contains list of files matching action files ex: reposyncer_action.yml
+      if(fileName == 'action.yaml'|| fileName == 'action.yml'){
+        core.info(`Found action in ${repoName}/${element}`)
+        // Get Forked from Info for the repo
+        const { data: repoinfo } = await client.rest.repos.get({
+          owner: repoOwner,
+          repo: repoName
+        })
+        let parentinfo = ''
+        if (repoinfo.parent?.full_name) {
+          parentinfo = repoinfo.parent.full_name
+        }
 
-      for (let index = 0; index < Object.keys(searchResultforRepository.data.items).length; index++) {
-        var element = searchResultforRepository.data.items[index].path;
+        // Get File content
         const { data: yaml } = await client.rest.repos.getContent({
-          owner: repo.owner,
-          repo: repo.name,
+          owner: repoOwner,
+          repo: repoName,
           path: element
         })
         if ('name' in yaml && 'download_url' in yaml) {
           result.name = yaml.name
-          result.repo = repo.name
+          result.repo = repoName
           result.forkedfrom = parentinfo
           if (yaml.download_url !== null) {
             result.downloadUrl = yaml.download_url
           }
         }
-      }
-      return result
-    }
 
-    core.info(`No actions found in repository: ${repo.name}`)
-    return null
-  }
-
-  return result
-}
-
-async function enrichActionFiles(
-  client: Octokit,
-  actionFiles: Content[]
-): Promise<Content[]> {
-  for (const action of actionFiles) {
-    // download the file in it and parse it
-    if (action.downloadUrl !== null) {
-      const { data: content } = await client.request({ url: action.downloadUrl })
-
-      // try to parse the yaml
-      try {
-        const parsed = YAML.parse(content)
-        const defaultValue = "Undefined" // Default value when json field is not defined
-        action.name = parsed.name ? parsed.name : defaultValue
-        action.author = parsed.author ? parsed.author : defaultValue
-        action.description = parsed.description ? parsed.description : defaultValue
-      } catch (error) {
-        // this happens in https://github.com/gaurav-nelson/github-action-markdown-link-check/blob/9de9db77de3b29b650d2e2e99f0ee290f435214b/action.yml#L9
-        // because of invalid yaml
-        console.log(
-          `Error parsing action file in repo [${action.repo}] with error:`
-        )
-        console.log(error)
-        console.log(
-          `The parsing error is informational, seaching for actions has continued`
-        )
+        actions.push(result)
       }
     }
   }
```

Comment:
I don't see a replacement for this code, so why is this not needed anymore?

Reply:
Looks like some issue with diff: 
line is there in https://github.com/vineeth-asml/load-available-actions/blob/fdaa5b24084b15b023f4efd84bcf363850c2c528/src/main.ts#L112